java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageSerializer.java

```diff
@@ -381,30 +473,29 @@ public static ArrowDictionaryBatch deserializeDictionaryBatch(ReadChannel in,
   /**
    * Deserialize a message that is either an ArrowDictionaryBatch or ArrowRecordBatch.
    *
-   * @param reader Interface to read messages from
-   * @param alloc Allocator for message data
+   * @param reader MessageReader to read a sequence of messages
    * @return The deserialized record batch
    * @throws IOException if the message is not an ArrowDictionaryBatch or ArrowRecordBatch
    */
-  public static ArrowMessage deserializeMessageBatch(MessageReader reader, BufferAllocator alloc) throws IOException {
-    Message message = reader.readNextMessage();
-    if (message == null) {
+  public static ArrowMessage deserializeMessageBatch(MessageReader reader) throws IOException {
```

Comment:
I think this is the only public method here that takes `MessageReader` as args? Seems a bit weird, any particular reason for that?

Reply:
Yeah, this doesn't really fit with the rest of the APIs and I didn't want to change it too much because I wasn't sure if someone relied on it.  @elahrvivaz , do you by chance use this one?

The problem with using `MessageReader` as an arg in the other functions is that they deserialize only a specific message type, e.g. `deserializeDictionaryBatch`, and `MessageReader` is not aware of any order of the messages - that is the job of `ArrowStreamReader`.  So to me it makes more sense to have either a `ReadChannel` or raw message data as input to deserialize.

Reply:
We only use the top-level reader API, so the internals don't matter much to me. Basically just `reader.loadNextBatch` and then use the populated `VectorSchemaRoot`